includes/class-wc-checkout.php

```diff
@@ -825,12 +827,20 @@ protected function validate_posted_data( &$data, &$errors ) {
 	 * @since  3.0.0
 	 * @param  array    $data   An array of posted data.
 	 * @param  WP_Error $errors Validation errors.
+	 * @throws Exception Invalid country code in billing or shipping address.
 	 */
 	protected function validate_checkout( &$data, &$errors ) {
 		$this->validate_posted_data( $data, $errors );
 		$this->check_cart_items();
 
-		if ( empty( $data['woocommerce_checkout_update_totals'] ) && empty( $data['terms'] ) && ! empty( $_POST['terms-field'] ) ) { // WPCS: input var ok, CSRF ok.
+		$billing_country = WC()->customer->get_shipping_country();
+		if ( ! WC()->countries->country_exists( $billing_country ) ) {
+			/* translators: %s: ISO 3166-1 alpha-2 country code. */
+			throw new Exception( sprintf( __( "'%s' is not a valid country code", 'woocommerce' ), $billing_country ) );
```

Comment:
instead of throwing immediately, should we be doing `$errors->add( $key . '_validation', ... );` to be consistent with other usages in this class?